### ISSUE TITLE
Unexpected behavior with rate limits using the set-style API

### ISSUE THREAD

**Issue by djannot** [solo-io/gloo#4095]
**Describe the bug**
Here are the results I get vs what I expect:
```
curl $(glooctl proxy url)/get -H 'x-api: voice' -H 'x-account-id: test_account' -v
# Never got 429. Expect to get it after 5 calls.
curl $(glooctl proxy url)/get -H 'x-api: voice' -H 'x-account-id: test_account' -H 'x-from-number: 1234567890' -v
# Got 429 after 1 call. Expect to get it after 2 calls.
curl $(glooctl proxy url)/get -H 'x-api: voice' -H 'x-account-id: test_account' -H 'x-from-number: 1111111111' -v
# Got 429 after 5 call. Expect to never get it.
```

**To Reproduce**
Here is the yaml I used:
```
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
  name: my-rate-limit-policy
  namespace: gloo-system
spec:
    raw:
        setDescriptors:
        -   simpleDescriptors:
            -   key: api
                value: voice
            -   key: account_id
                value: test_account
            rateLimit:
                requestsPerUnit: 5
                unit: MINUTE
            alwaysApply: true
        -   simpleDescriptors:
            -   key: api
                value: voice
            -   key: accountid
                value: test_account
            -   key: from_number
                value: "1234567890"
            rateLimit:
                requestsPerUnit: 2
                unit: MINUTE
            alwaysApply: true
        rateLimits:
        -   setActions:
            -   requestHeaders:
                    descriptorKey: api
                    headerName: x-api
            -   requestHeaders:
                    descriptorKey: accountid
                    headerName: x-account-id
            -   requestHeaders:
                    descriptorKey: fromnumber
                    headerName: x-from-number
```

**Additional context**
- Gloo Edge version: 1.6.1

**Comment by kdorosh** [comment]
Combine the above yaml with:
```yaml
apiVersion: gateway.solo.io/v1
kind: VirtualService
metadata:
  name: test-static
  namespace: gloo-system
spec:
  virtualHost:
    domains:
    - foo
    options:
      rateLimitConfigs:
        refs:
        - name: my-rate-limit-policy
          namespace: gloo-system
    routes:
    - matchers:
      - prefix: /
      options:
        autoHostRewrite: true
      routeAction:
        single:
          upstream:
            name: json-upstream
            namespace: gloo-system
```

and

```yaml
apiVersion: gloo.solo.io/v1
kind: Upstream
metadata:
  name: json-upstream
  namespace: gloo-system
spec:
  static:
    hosts:
    - addr: jsonplaceholder.typicode.com
      port: 80
```

to create an env to reproduce from clean install

**Comment by kdorosh** [comment]
Also, fix the rate limit config yaml, it is malformed (some of the keys are inconsistent with underscores vs spaces):

```yaml
apiVersion: ratelimit.solo.io/v1alpha1
kind: RateLimitConfig
metadata:
  name: my-rate-limit-policy
  namespace: gloo-system
spec:
  raw:
    rateLimits:
    - setActions:
      - requestHeaders:
          descriptorKey: api
          headerName: x-api
      - requestHeaders:
          descriptorKey: accountid
          headerName: x-account-id
      - requestHeaders:
          descriptorKey: from_number
          headerName: x-from-number
    setDescriptors:
    - alwaysApply: true
      rateLimit:
        requestsPerUnit: 5
        unit: MINUTE
      simpleDescriptors:
      - key: api
        value: voice
      - key: accountid
        value: test_account
    - alwaysApply: true
      rateLimit:
        requestsPerUnit: 2
        unit: MINUTE
      simpleDescriptors:
      - key: api
        value: voice
      - key: accountid
        value: test_account
      - key: from_number
        value: "1234567890"
```

**Comment by kdorosh** [comment]
FYI, w.r.t. the third case.
```
curl $(glooctl proxy url)/get -H 'x-api: voice' -H 'x-account-id: test_account' -H 'x-from-number: 1111111111' -v
# Got 429 after 5 call. Expect to never get it.
```

This is actually desired behavior. This request matches the following simple descriptor and thus has its limit:
```yaml
    - alwaysApply: true
      rateLimit:
        requestsPerUnit: 5
        unit: MINUTE
      simpleDescriptors:
      - key: api
        value: voice
      - key: accountid
        value: test_account
```

**Comment by kdorosh** [comment]
Fix has been merged into master in closed source, reopening until the fix has been backported (and then soon released)

**Comment by kdorosh** [comment]
The fix has been released in 1.6.5
